sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/execution_dispatcher.py

```diff
@@ -107,12 +108,30 @@ async def _create_pipelined_execution_context(self, query_execution_info):
                 raise CosmosHttpResponseError(StatusCodes.BAD_REQUEST,
                                   "Cross partition query only supports 'VALUE <AggregateFunc>' for aggregates")
 
-        execution_context_aggregator = multi_execution_aggregator._MultiExecutionContextAggregator(self._client,
+        # throw exception here for vector search query without limit filter or limit > max_limit
+        if query_execution_info.get_non_streaming_order_by():
+            total_item_number = query_execution_info.get_top() or query_execution_info.get_limit()
```

Comment:
Any chance that `get_top()` could return `0` as a valid output? If so, it would get ignored by this statement (whereas `get_limit()` returning `0` would be accepted)

Reply:
Actually I can't see where `total_item_number` is used? Or is this code just to validate it?

Reply:
it's just to validate that a filter is being provided by the user - if they provide 0 for either number (meaning they want 0 results) then the behavior should give them that. But no, in general it will always return an int that is set or None as the default.